app/controllers/assessment/handin.rb

```diff
@@ -344,7 +344,7 @@ def saveHandin(sub)
 
   def get_handin
     if @assessment.nil?
-      @assessment = @course.assessments.find(params[:assessment_id])
+      @assessment = @course.assessments.find_by_name(params[:assessment_id])
```

Comment:
I'm not certain this check is necessary.  I think assessment will always be defined before this function is called, but you can double check.